### PR TITLE
Proposed fix for Latex sub/super-scripts in legends

### PR DESCRIPTION
**Pull request by t-bltg**
Fixes https://github.com/JuliaPlots/Plots.jl/issues/3558:
1. Fix height computation by retaining the maximum legend height, for each serie;
2. Allow changing the height factor via the `extra_kwargs` mecanism.

Discussion:
Change [1.] does modify some reference images, but I think the fix is justified. Only applying [2.] allows fixing the OP issue, without changing the default legend sizes.

MWE
-------
```julia
using LaTeXStrings
using Random; Random.seed!(1234)
using Plots; gr()

main() = begin
  y1, y2 = rand(100), rand(100)
  histogram(y1, ylims=(0, 50), label=L"\left\Vert\beta_{1}^{2}-\beta_{2}^{2}\right\Vert", extra_kwargs=Dict(:subplot=>Dict(:legend_hfactor=>1.25)))
  histogram!(y2, label=L"\left\Vert\beta_{2}^{2}-\beta_{3}^{2}\right\Vert", legendfontsize=10)
  png("foo")
end

main()
```

With PR
-----------
Without using `extra_kwargs`:
pr3

Using `:legend_hfactor=>1.25`:
pr3

Without PR
---------------
pr3

Default behavior modification (example # 23)
==================================

With PR
-----------
ref23

Without PR
---------------
ref23_old

### CODE REVIEW

**Review by daschw — APPROVED** [review]
Looks great!

**Comment by t-bltg** [comment]
@ptoche, is this approach ok for you ?

File: src/backends/gr.jl [hunk 1]
```diff
@@ -1162,25 +1162,25 @@ function gr_legend_pos(theta::Real, leg, viewport_plotarea; axisclearance=nothin
 end
 
 function gr_get_legend_geometry(viewport_plotarea, sp)
-    legendn = 0
-    legendw = 0
+    legendn = legendw = 0; dy = 0.
```
Comment (BeastyBlacksmith) [inline src/backends/gr.jl:1165]:
```suggestion
    legendn = legendw = 0; dy = 0
```

File: src/backends/gr.jl [hunk 2]
```diff
@@ -1197,7 +1197,8 @@ function gr_get_legend_geometry(viewport_plotarea, sp)
     x_legend_offset = (viewport_plotarea[2] - viewport_plotarea[1]) / 30
     y_legend_offset = (viewport_plotarea[4] - viewport_plotarea[3]) / 30
 
-    dy = gr_point_mult(sp) * sp[:legendfontsize] * 1.75
+    dy *= get(sp[:extra_kwargs], :legend_hfactor, 1.)
```
Comment (BeastyBlacksmith) [inline src/backends/gr.jl:1200]:
```suggestion
    dy *= get(sp[:extra_kwargs], :legend_hfactor, 1)
```
If these don't have to be floats for a good reason, integer literals are preferred.

**Comment by ptoche** [comment]
This looks great, thanks for putting this together! :-)

**Comment by t-bltg** [comment]
You can use [`legend_hfactor`](https://github.com/t-bltg/Plots.jl/blob/293b43013bd0fd9ac022c706025b877c86a9ca5a/src/backends/gr.jl#L1200) > 1 to fine tune the spacing.

**Comment by lmiq** [comment]
That does not seem to have any effect on my plot. Am I doing something wrong?

image

**Pull request merged**